pkg/ops/rawDiskGeneration.go

```diff
@@ -979,3 +980,39 @@ func (r *RawImage) FinalizeImage(image string) error {
 	}
 	return nil
 }
+
+// CopyAlpineShimAndGrub copies the grub.efi to the shim place and also to the grub place
+// As alpine does not provide a shim, we need to copy the grub.efi to the shim place
+// so we have the same behaviour as in other flavors
+func (r *RawImage) CopyAlpineShimAndGrub(arch, target string) error {
+	var err error
+	grubPath := filepath.Join("/efi", arch, constants.EfiBootPath, "bootx64.efi")
+	targetPath := filepath.Join(target, constants.EfiBootPath, "bootx64.efi")
+
+	if arch == "arm64" {
+		grubPath = filepath.Join("/efi", arch, constants.EfiBootPath, "bootaa64.efi")
+		targetPath = filepath.Join(target, constants.EfiBootPath, "bootaa64.efi")
+	}
+
+	err = utils.CopyFile(
+		r.config.Fs,
+		grubPath,
+		targetPath,
+	)
+	if err != nil {
+		return fmt.Errorf("could not write file %s at dir %s from %s", "bootx64.efi", target, filepath.Join("/efi", constants.EfiBootPath, "grub.efi"))
```

Comment:
This messages is misleading. `grub.efi` is not the source file you are using above (`bootx64.efi` or `bootaa64.efi`)